intake/cli/server/server.py

```diff
@@ -65,15 +70,23 @@ def _make_cache_callback(self, callback, idle_time):
 class ServerInfoHandler(tornado.web.RequestHandler):
     def initialize(self, catalog):
         self.catalog = catalog
+        auth = conf.get('auth', 'intake.auth.base.BasicAuth')
+        self.auth = get_auth_class(auth['class'], *auth.get('args', tuple()),
+                                   **auth.get('kwargs', {}))
 
     def get(self):
-        sources = []
-        for _, name, source in self.catalog.walk():
-            info = source.describe()
-            info['name'] = name
-            sources.append(info)
-
-        server_info = dict(version='0.0.1', sources=sources)
+        head = self.request.headers
+        if self.auth.allow_connect(head):
+            sources = []
+            for _, name, source in self.catalog.walk():
+                if self.auth.allow_access(head, source):
+                    info = source.describe()
+                    info['name'] = name
+                    sources.append(info)
+
+            server_info = dict(version='0.0.1', sources=sources)
+        else:
+            server_info = dict(version='0.0.1', sources=[])
```

Comment:
If `allow_connect` fails, we should return 403 (same as failure of `allow_access` below), otherwise the client won't be able to tell the difference between overall authorization failure and an empty remote catalog.